### PR TITLE
Add `uv init --virtual`

### PR DESCRIPTION
**Pull request by j178**
## Summary

Add `uv init --virtual` to create an explicit virtual workspace.

Relates to #5338

### CODE REVIEW

File: crates/uv-workspace/src/workspace.rs [hunk 1]
```diff
@@ -996,7 +996,7 @@ fn check_nested_workspaces(inner_workspace_root: &Path, options: &DiscoveryOptio
             if !is_excluded {
                 warn_user!(
                     "Nested workspaces are not supported, but outer workspace includes existing workspace: `{}`",
-                    pyproject_toml_path.user_display().cyan(),
+                    outer_workspace_root.simplified_display().cyan(),
```
Comment (j178) [inline crates/uv-workspace/src/workspace.rs:999]: Running from the current directory, this message is "Nested workspaces are not supported, but outer workspace includes existing workspace: ``", seems not clear.

File: crates/uv/src/commands/project/init.rs [hunk 2]
```diff
@@ -77,6 +75,94 @@ pub(crate) async fn init(
...
+        }
+    }
+
+    Ok(ExitStatus::Success)
+}
+
+fn init_virtual_workspace(path: &Path, isolated: bool) -> Result<()> {
+    // Check nested workspaces.
+    if !isolated {
+        check_nested_workspaces(path, &DiscoveryOptions::default());
```
Comment (j178) [inline crates/uv/src/commands/project/init.rs:138]: Should we warn or abort if nested workspace detected?

**Review by charliermarsh — APPROVED** [review]

File: crates/uv/src/commands/project/init.rs (hunk 2 above)
Comment (charliermarsh) [reply crates/uv/src/commands/project/init.rs:139]: I think warn is ok.

File: crates/uv-workspace/src/workspace.rs (hunk 1 above)
Comment (charliermarsh) [reply crates/uv-workspace/src/workspace.rs]: Good call.

**Pull request merged**